apps/server/src/modules/school/service/school.service.ts

```diff
@@ -31,7 +31,12 @@ export class SchoolService {
 	): Promise<OauthMigrationDto> {
 		const schoolDo: SchoolDO = await this.schoolRepo.findById(schoolId);
 		if (oauthMigrationPossible != null) {
-			schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
+			if (!schoolDo.oauthMigrationFinished && !schoolDo.oauthMigrationPossible) {
```

Comment:
Maybe you can take move it to a private method with a understandable name like: isSystemMigrationStarted

Reply:
As there are no more than 2 things happening at once, i think this might be an overkill moving it to private functions. Especially when we have a timely refactoring in mind changing those "early stage variable" names into something understandable. Also: this is a bug fix ;) But i will add the lines in the spec file for readability ^^

Reply:
setMigration reads like a mix of orechstration and logic operations. I think to clean it up over a private method make sense. 
It include also multiple times the operations boolean ? new Date() : undefined. I want to like if this also move out in a private function with good naming.

Reply:
should be more readable now :)